internal/config/config.go

```diff
@@ -331,60 +253,65 @@ func Parse(bs []byte, validate Validate) (*Config, error) {
 	return cfg, nil
 }
 
-func parsePeer(p peer) (*Peer, error) {
-	if p.MyASN == 0 {
+func peerFromCR(p metallbv1beta1.BGPPeer) (*Peer, error) {
+	if p.Spec.MyASN == 0 {
 		return nil, errors.New("missing local ASN")
 	}
-	if p.ASN == 0 {
+	if p.Spec.ASN == 0 {
 		return nil, errors.New("missing peer ASN")
 	}
-	if p.ASN == p.MyASN && p.EBGPMultiHop {
+	if p.Spec.ASN == p.Spec.MyASN && p.Spec.EBGPMultiHop {
 		return nil, errors.New("invalid ebgp-multihop parameter set for an ibgp peer")
 	}
-	ip := net.ParseIP(p.Addr)
+	ip := net.ParseIP(p.Spec.Address)
 	if ip == nil {
-		return nil, fmt.Errorf("invalid peer IP %q", p.Addr)
+		return nil, fmt.Errorf("invalid peer IP %q", p.Spec.Address)
 	}
-	holdTime, err := parseHoldTime(p.HoldTime)
-	if err != nil {
-		return nil, err
+	holdTime := p.Spec.HoldTime.Duration
+	if holdTime == 0 { // TODO move the defaults at CRD level if they are not already
+		holdTime = 90 * time.Second
 	}
-	keepaliveTime, err := parseKeepaliveTime(holdTime, p.KeepaliveTime)
+	err := validateHoldTime(holdTime)
 	if err != nil {
 		return nil, err
 	}
+	keepaliveTime := p.Spec.KeepaliveTime.Duration
+	if keepaliveTime == 0 {
+		keepaliveTime = holdTime / 3
+	}
+
 	// keepalive must be lower than holdtime
 	if keepaliveTime > holdTime {
-		return nil, fmt.Errorf("invalid keepaliveTime %q", p.KeepaliveTime)
+		return nil, fmt.Errorf("invalid keepaliveTime %q", p.Spec.KeepaliveTime)
 	}
 	port := uint16(179)
-	if p.Port != 0 {
-		port = p.Port
+	if p.Spec.Port != 0 {
+		port = p.Spec.Port
```

Comment:
should we add port default value=179 to the CRD instead of this?

Reply:
See the comment above